### PR TITLE
snakecase whichbonds arg and update tests

### LINKED ISSUE #160: change whichbonds arg naming convention
better to use the snakecase naming convention consistently in the code

### PR DESCRIPTION
**Pull request by naik-aakash**
Closes #160 

# Changes
1.  Updated the which bonds arg and the accepted input string. 
2.  adapted existing tests

### CODE REVIEW

**Comment by naik-aakash** [comment]
@JaGeo, PR should be ready for review 😄

File: lobsterpy/cohp/analyze.py [hunk 1]
```diff
@@ -187,11 +187,11 @@ def setup_env(self):
 
         # determine cations and anions
         try:
-            if self.whichbonds == "cation-anion":
+            if self.which_bonds == "cation_anion":
```
Comment (JaGeo) [inline lobsterpy/cohp/analyze.py:190]: `cation-anion` is not a parameter. Just a string that we pass.

File: lobsterpy/featurize/core.py [hunk 2]
```diff
@@ -208,11 +208,7 @@ def get_lobsterpy_cba_dict(path_to_lobster_calc: str, bonds: str) -> dict:
         icohplist_path = req_files_lobsterpy.get("icohplist_path")
 
         which_bonds = bonds
-
-        if which_bonds == "all":
-            bond_type = "all_bonds"
-        elif which_bonds == "cation-anion":
-            bond_type = "cation_anion_bonds"
+        bond_type = f"{which_bonds}_bonds"
```
Comment (JaGeo) [inline lobsterpy/featurize/core.py:211]: I personally think that this is the problem. Not the string.

File: lobsterpy/featurize/test/test_core.py [hunk 3]
```diff
@@ -486,27 +486,27 @@ def test_lobsterpy_featurize_exception(self):
...
-                path_to_lobster_calc=TestDir / "TestData/C/", bonds="cation-anion"
+                path_to_lobster_calc=TestDir / "TestData/C/", bonds="cation_anion"
             )
 
             _ = self.featurize_C_cation_anion.get_df()
 
         self.assertEqual(
             err.exception.__str__(),
-            "No cation-anion bonds detected for C structure. "
+            "No cation_anion bonds detected for C structure. "
```
Comment (JaGeo) [inline lobsterpy/featurize/test/test_core.py:509]: Doen't make much sense to write it in this way. I think the changes are not completely thought through yet.

**Review by JaGeo — CHANGES_REQUESTED** [review]
@naik-aakash Thanks! I think I would recommend thinking about the difference between the parameter name and a string. Usage of snake case is only recommended for the parameters.

File: lobsterpy/cohp/analyze.py (hunk 1 above)
Comment (naik-aakash) [reply lobsterpy/cohp/analyze.py:190]: But the string is the thing we change in atomate2 and it was pointed out that is the weird fix? What exactly I should do in atomate2 then? I do not get it 😢

File: lobsterpy/featurize/core.py (hunk 2 above)
Comment (naik-aakash) [reply lobsterpy/featurize/core.py:211]: But in atomate2, we already have the snake case for this parameter. Can you please be specific what is the problem there?

File: lobsterpy/cohp/analyze.py (hunk 1 above)
Comment (JaGeo) [reply lobsterpy/cohp/analyze.py:190]: Maybe, we don't need the change in atomate2.

**Comment by naik-aakash** [comment]
Hi @JaGeo, if this is fine, can you please merge it as well? The only change is which_bonds arg in snake case and rest is same as before 😃

**Review by JaGeo — APPROVED** [review]

**Pull request merged**